crates/task-impls/src/consensus/mod.rs

```diff
@@ -1050,6 +1051,9 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     }
                 }
 
+                // Clear the payload commitment and metadata
+                self.payload_commitment_and_metadata = None;
+
```

Comment:
I don't think we want to clear this here because we could have set the option before a view change event and then clear it before we can propose with it

Reply:
Yeah, I suspected this would cause problems -- changed to a clear based on view number!